test/regExpTest.coffee

```diff
@@ -0,0 +1,71 @@
+expect = require('chai').expect
+
+describe 'regExp matches URI', ->
+  regExp = new RegExp("^(http[s]?:\\/\\/(www\\.)?|ftp:\\/\\/(www\\.)?|www\\.){1}([0-9A-Za-z-\\.@:%_\+~#=]+)+((\\.[a-zA-Z]{2,3})+)(/(.)*)?(\\?(.)*)?", "i")
```

Comment:
If you leave the above, you need to test every part of it. Things not tested include support for username and passwords in the URI, anchors(#), home path on ftp (~), etc. 

Also the `i` flag at the end indicates case insensitive, which means all the `[A-Za-z]` can simply be `[a-z]`.